sammba/modality_processors/perfusion_fair.py

```diff
@@ -0,0 +1,387 @@
+    # *_ptbl.txt file has a specific format created by
+    # sammba.io_conversions.dcm_to_nii. if someone supplies their own file
+    # name, it would mean they changed the *_ptbl.txt name (why?) or they have
+    # their own formatted text file which this function would not be able to
+    # process
+    
+    bfx = nii_fname.split('_')[-1].split('.')[0]
+    ptbl = pd.read_table(nii_fname.replace(bfx + '.nii.gz', bfx + '_ptbl.txt'))
+    ti = ptbl.TI[ptbl.slice == 1][ptbl.FC == 'Selective Inversion'].tolist()
+    long_ti = ptbl.TI[ptbl.slice == 1].tolist()
+    fc = ptbl.FC[ptbl.slice == 1].tolist()
+    picker_sel = [n for n,x in enumerate(fc) if x == 'Selective Inversion']
+    picker_nonsel = [n for n,x in enumerate(fc) if x == 'Non-selective Inversion']
+    return {'TI':ti, 'long_TI':long_ti, 'FC':fc,
+            'picker_sel':picker_sel, 'picker_nonsel':picker_nonsel}
+
+
+# the perfusion fluid-attenuated inversion-recovery function
+# the jacobian should be callable but I do not know enough maths to create it
+# pars = parameters
+def _fair_t1_func(pars, s0, ti):
+    return pars[0] + np.absolute(pars[1] * (1 - 2 * np.exp(-ti / pars[2]))) - s0
+
+
+def fair_t1_fit(s0, ti, t1_guess):
+    """
+    Fit the perfusion FAIR equation:
+        
+        signal = bias + abs(M0 * (1 - 2 * (exp(-TI/T1))))
+        
+    using scipy.optimize.least_squares (default Trust Region Reflective
+    algorithm). Bias, M0 and T1 are all estimated with a zero lower bound.
+    Starting values for bias (zero) and M0 (mean of input signals) are
+    calculated automatically; T1 has to be supplied, with the mean of TIs
+    being a good guess.
+    
+    Parameters
+    ----------
+    s0 : numpy array of int or float
+        The aquired signals.
+        
+    ti : list of int or float
+        The inversion times. Must have the same length as s0.
+        
+    t1_guess : int or float
+        An initial starting value for T1. The mean of TIs is a good guess.
+
+    Returns
+    -------
+    class scipy.optimize.OptimizeResult    
+    
+    Notes
+    ----- 
+    Levenberg-Marquardt (method='lm'), the usual literature algorithm, cannot
+    be used here as the implementation in scipy does not accept bounds. From
+    experience with the R MINPACK implementation (minpack.lm function nlsLM),
+    these bounds are necessary to assure good fitting. The lmfit python package
+    does implement bounds for MINPACK Levenberg-Marquardt (method=leastsq;
+    default), though the results appear to be worse than for TRR
+    (method=least_squares), whose results bizarrely look closer to those of R
+    minpack.lm nlsLM (to confirm), and are of course identical to
+    scipy.optimize.least_squares with method='trf' (default).
+    """
+    
+    return ls(_fair_t1_func, np.array([0, np.mean(s0), t1_guess]),
+              bounds=([0, 0, 0], np.inf), args=(s0, np.array(ti)))
+
+
+def perf_fair_fitter(s0, t1_blood, ti, t1_guess, picker_sel, picker_nonsel,

```

Comment:
all instances of fitter are now fit

Reply:
`fit` is somehow vague for me. Can't you find another name that gives more information about what the function does, e.g. `compute_cbf`.
Also, I think you don't really need to prefix with `perf_fair`, since your module is named perfusion_fair